### PR TITLE
warnings: fix missing None in existing hook & add some docs

### PR DESCRIPTION
**Pull request by bluetech**
This covers some loose ends from #7255 and fixes the issue reported later in #4049.

### CODE REVIEW

**Comment by bluetech** [comment]
cc @gnikonorov. Wanted to submit this before we forget about this topic.

File: doc/en/deprecations.rst [hunk 1]
```diff
@@ -20,6 +20,17 @@ Below is a complete list of all pytest features which are considered deprecated.
 :ref:`standard warning filters <warnings>`.
 
 
+The ``pytest_warning_captured`` hook
```
Comment (gnikonorov) [inline doc/en/deprecations.rst:23]: I didn't realize we had a deprecations file. Thanks @bluetech

File: src/_pytest/warnings.py [hunk 2]
```diff
@@ -112,7 +112,12 @@ def catch_warnings_for_item(config, ihook, when, item):
 
         for warning_message in log:
             ihook.pytest_warning_captured.call_historic(
```
Comment (gnikonorov) [inline src/_pytest/warnings.py:114]: Do we usually update deprecated hooks? Asking for my own knowledge. I wasn't sure when I made the change for `pytest_warning_recorded`

Comment (bluetech) [reply src/_pytest/warnings.py:114]: I think it makes sense to update deprecated hooks if it's a simple bugfix like this.

**Pull request merged**